### DISCUSSION TITLE
Loading Toast

### DISCUSSION THREAD

**Discussion by lucasaarcoverde** [vtex/shoreline#827] [Design RFCs]
> This issue was originally opened by @gtkatakura on **28 Oct 2021**

# Feature Request

- [x] I've read and understood the [Contributing guidelines](https://admin-ui-docs.vercel.app/engineering/contributing/) and have done my best effort to follow them.
- [x] I've searched for any related issues and avoided creating a duplicate issue.

## Is this feature request related to an existing component? Please describe.

I would like to have the option to set a custom icon or element to be rendered on the `showToast`.
Today we have a case in which we would like to use the toast to show a message that is waiting for something to happen. The message is `Verifying access...`, and would be good to show a spinner instead of the info icon to the user understand that something is happening.

**Comment by matheusps** [comment]
Implemented on the Toast/promise.
